File: NAP_II_JAVA/ClasseECommerce/ECommerce.java
import java.util.ArrayList;
import java.util.List;

public class ECommerce {
    private List<Produto> produtos;
    private List<Cliente> clientes;
    private List<Pedido> pedidos;
    private List<Transportadora> transportadoras;

    // Construtor não padrão
    public ECommerce() {
        this.produtos = new ArrayList<>();
        this.clientes = new ArrayList<>();
        this.pedidos = new ArrayList<>();
        this.transportadoras = new ArrayList<>();
    }

    // Getters
    public List<Produto> getProdutos() {
        return produtos;
    }

    public List<Cliente> getClientes() {
        return clientes;
    }

    public List<Pedido> getPedidos() {
        return pedidos;
    }

    public List<Transportadora> getTransportadoras() {
        return transportadoras;
    }

    // Sobrescrevendo toString
    @Override
    public String toString() {
        return "ECommerce{" +
                "produtos=" + produtos +
                ", clientes=" + clientes +
                ", pedidos=" + pedidos +
                ", transportadoras=" + transportadoras +
                '}';
    }
}

File: NAP II - POO/Python/UFRAcommerce/ecommerce.py
class ECommerce:
    def __init__(self):
        self.produtos = []
        self.clientes = []
        self.pedidos = []
        self.transportadoras = []

    def pagar(self, pedido, valor, voucher=None):
        desconto = 0
        if voucher:
            desconto = pedido.total * 0.1
        if valor >= (pedido.total - desconto):
            return True
        return False

    def cadastrar(self, cliente):
        self.clientes.append(cliente)

    def comprar(self, produto):
        self.produtos.append(produto)

    def credenciar(self, transportadora):
        self.transportadoras.append(transportadora)

    def __str__(self):
        return f"ECommerce(produtos={len(self.produtos)}, clientes={len(self.clientes)}, pedidos={len(self.pedidos)})"
